### PR TITLE
Api 33 support

### PR DESCRIPTION
**Pull request by connyduck**
Pretty basic Api level upgrade, fixing of new deprecations, handling notification permission

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/MainActivity.kt [hunk 1]
```diff
@@ -267,6 +272,33 @@ class MainActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidInje
...
+                        else -> {
+                            finish()
+                        }
+                    }
+                }
+            }
+        )
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/MainActivity.kt:296]: This logic very basic but I think it is enough? Android will not show the dialog if a user denied the Permission more than twice anyway.

**Comment by connyduck** [comment]
Will do the native language switching thing in another pull request, maybe Appcompat 1.6.0 is stable until then

File: app/src/main/java/com/keylesspalace/tusky/TabPreferenceActivity.kt [hunk 2]
```diff
@@ -74,6 +74,12 @@ class TabPreferenceActivity : BaseActivity(), Injectable, ItemInteractionListene
 
     private val hashtagRegex by lazy { Pattern.compile("([\\w_]*[\\p{Alpha}_][\\w_]*)", Pattern.CASE_INSENSITIVE) }
 
+    private val onFabDismissedCallback = object: OnBackPressedCallback(false) {
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/TabPreferenceActivity.kt]: why `false`?

File: app/src/main/java/com/keylesspalace/tusky/components/preference/PreferencesActivity.kt [hunk 3]
```diff
@@ -148,20 +156,6 @@ class PreferencesActivity :
         overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
     }
 
-    override fun onBackPressed() {
-        /* Switching themes won't actually change the theme of activities on the back stack.
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/components/preference/PreferencesActivity.kt:152]: we've lost this comment

File: app/src/main/java/com/keylesspalace/tusky/MainActivity.kt (hunk 1 above)
Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/MainActivity.kt:296]: Seems fine

File: app/src/main/java/com/keylesspalace/tusky/TabPreferenceActivity.kt (hunk 2 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/TabPreferenceActivity.kt]: So it is inactive. It gets activated when the FAB is extended and on back press collapses it. When the FAB is not extended I want the default behavior on back press = close the activity.

File: app/src/main/java/com/keylesspalace/tusky/components/preference/PreferencesActivity.kt (hunk 3 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/components/preference/PreferencesActivity.kt:152]: added it back

**Comment by K900** [comment]
Looks like this change broke picking/saving files on Android 13 (possibly only QPR1 beta? Can't test right now). Attempting to save or post any file just gives the "permission required" snackbar with no OS permission prompt or anything. The stable version built against API 31 works.

**Comment by connyduck** [comment]
@K900 please create an issue so I don't forget to check that

**Comment by K900** [comment]
Filed https://github.com/tuskyapp/Tusky/issues/2801

**Pull request merged**